ocaml/xapi/session_check.ml

```diff
@@ -53,9 +53,21 @@ let check ~intra_pool_only ~session_id ~action =
           if (not pool) && not (Pool_role.is_master ()) then
             raise Non_master_login_on_slave ;
           if Pool_role.is_master () then
-            Db_actions.DB_Action.Session.set_last_active ~__context
-              ~self:session_id
-              ~value:(Xapi_stdext_date.Date.of_float (Unix.time ()))
+            (* before updating the last_active field, check if the field has been
+               already updated recently. This avoids holding the database lock too often.*)
+            let n = Xapi_stdext_date.Date.of_float (Unix.time ()) in
+            let last_active =
+              Db_actions.DB_Action.Session.get_last_active ~__context
+                ~self:session_id
+            in
+            let threshold_time = Xapi_stdext_date.Date.diff n last_active in
+            if
+              Ptime.Span.compare threshold_time
```

Comment:
I would prefer the `is_earlier`, `is_later` functions because they are more obvious to read.

Reply:
https://github.com/xapi-project/xen-api/pull/5976/commits/b95ddb7c1e315de059fcc557cb9280956cd81083 I agree. I a looked over the implementation of `diff` and it does the same conversion. So, I did it explicitly and use `is later`. The error path should not be happening.